src/types/request.rs

```diff
@@ -256,25 +256,55 @@ impl PyRequest {
 
     pub fn json(&self, py: Python) -> PyResult<PyObject> {
         match self.body.as_ref(py).downcast::<PyString>() {
-            Ok(python_string) => match serde_json::from_str(python_string.extract()?) {
-                Ok(Value::Object(map)) => {
-                    let dict = PyDict::new(py);
-
-                    for (key, value) in map.iter() {
-                        let py_key = key.to_string().into_py(py);
-                        let py_value = match value {
-                            Value::String(s) => s.as_str().into_py(py),
-                            _ => value.to_string().into_py(py),
-                        };
+            Ok(python_string) => json_string_to_pyobject(py, python_string.extract()?),
+            Err(e) => Err(e.into()),
+        }
+    }
+}
 
-                        dict.set_item(py_key, py_value)?;
-                    }
+///serde_json not impl IntoPy<PyObject>  but orphan principle. so use this function
+fn json_string_to_pyobject(py: Python, json_string: &str) -> PyResult<PyObject> {
+    /// Converts a serde_json::Number to a Python object.
+    fn number_to_pyobject(num: serde_json::Number, py: Python) -> PyResult<PyObject> {
+        match num.as_u64() {
+            Some(u) => Ok(u.into_py(py)),
+            None => match num.as_i64() {
+                Some(i) => Ok(i.into_py(py)),
+                None => match num.as_f64() {
+                    Some(f) => Ok(f.into_py(py)),
+                    None => Err(PyErr::new::<pyo3::exceptions::PyTypeError, _>(
+                        "Invalid number format",
+                    )),
+                },
+            },
+        }
+    }
 
-                    Ok(dict.into_py(py))
+    /// Converts a serde_json::Value to a Python object.
+    fn value_to_pyobject(py: Python, value: serde_json::Value) -> PyResult<PyObject> {
+        match value {
+            Value::Null => Ok(().into_py(py)),
+            Value::Bool(b) => Ok(b.into_py(py)),
+            Value::Number(num) => number_to_pyobject(num, py),
+            Value::String(s) => Ok(s.into_py(py)),
+            Value::Array(array) => array
+                .into_iter()
+                .map(|v| value_to_pyobject(py, v))
+                .collect::<PyResult<Vec<PyObject>>>()
+                .map(|list| PyList::new(py, list).into()),
+            Value::Object(map) => {
+                let dict = PyDict::new(py);
+                for (key, value) in map.into_iter() {
+                    let py_key = key.to_string().into_py(py);
+                    let py_value = value_to_pyobject(py, value)?;
+                    dict.set_item(py_key, py_value)?;
                 }
-                _ => Err(PyValueError::new_err("Invalid JSON object")),
```

Comment:
it‘s make server 500 status ,I think it catch by python code.

Reply:
In the robyn.pyi file, the documentation for the function indicates that returning a ValueError is normal if the JSON is not valid.
```python
    def json(self) -> dict:
        """
        If the body is a valid JSON this will return the parsed JSON data.
        Otherwise, this will throw a ValueError.
        """
        pass
```